python/rmm/_lib/memory_resource.pyx

```diff
@@ -49,22 +49,30 @@ cdef class PoolMemoryResource(MemoryResource):
     def __cinit__(
             self,
             MemoryResource upstream,
-            size_t initial_pool_size=~0,
-            size_t maximum_pool_size=~0
+            initial_pool_size=None,
+            maximum_pool_size=None
     ):
+        cdef size_t c_initial_pool_size
+        cdef size_t c_maximum_pool_size
+        c_initial_pool_size = (
+            ~0 if initial_pool_size is None else initial_pool_size
+        )
+        c_maximum_pool_size = (
+            ~0 if maximum_pool_size is None else maximum_pool_size
+        )
```

Comment:
AFAICT there's no super clean way to handle this (e.g., the way @harrism would like to see as detailed in https://github.com/rapidsai/rmm/issues/497). Python needs to be able to handle when the user specifies the `maximum_pool_size` but not the `initial_pool_size`. There's no way to do that in C++ without explicitly specifying the sentinel value `~0` for `initial_pool_size` (is there?).

Reply:
I think we'd need another C++ overload or to change the C++ API to have a builder pattern or something similar. cc @harrism @jrhemstad

Reply:
Merging as the Python side is fine for now. We can move to a different strategy if desired in a follow up PR.

Reply:
@shwina you should access the default value from the C++ class.